src/rax.c

```diff
@@ -1380,19 +1397,15 @@ int raxIteratorNextStep(raxIterator *it, int noup) {
                 /* Try visiting the next child if there was at least one
                  * additional child. */
                 if (!it->node->iscompr && it->node->size > (old_noup ? 0 : 1)) {
-                    raxNode **cp = raxNodeFirstChildPtr(it->node);
-                    int i = 0;
-                    while (i < it->node->size) {
-                        debugf("SCAN NEXT %c\n", it->node->data[i]);
-                        if (it->node->data[i] > prevchild) break;
-                        i++;
-                        cp++;
-                    }
-                    if (i != it->node->size) {
+                    debugf("it->node->size: %d\n", it->node->size);
+                    if (it->child_offset + (old_noup ? 0 : 1) != it->node->size) {
```

Comment:
Look like `if (!it->node->iscompr && it->node->size > (old_noup ? 0 : 1))` and `if (it->child_offset + (old_noup ? 0 : 1) != it->node->size) ` can be combined to `if (!it->node->iscompr && it->child_offset + (old_noup ? 0 : 1) != it->node->size)`

Reply:
Nice one! Passes all tests. I'll commit and push.